output/output.go

```diff
@@ -119,3 +127,145 @@ func (e exporter) ExportSimulatorDiagnostics(deployDir, pth, name string) error
 
 	return nil
 }
+
+func (e exporter) ExportFlakyTestCases(xcResultPath string, useOldXCResultExtractionMethod bool) error {
+	testSummary, err := e.parseTestSummary(xcResultPath, useOldXCResultExtractionMethod)
+	if err != nil {
+		return fmt.Errorf("failed to parse test summary: %w", err)
+	}
+
+	flakyTestPlans := e.collectFlakyTestPlans(*testSummary)
+	if len(flakyTestPlans) == 0 {
+		return nil
+	}
+
+	return e.exportFlakyTestCases(flakyTestPlans)
+}
+
+func (e exporter) parseTestSummary(xcResultPath string, useOldXCResultExtractionMethod bool) (*model3.TestSummary, error) {
+	converter := xcresult3.Converter{}
+	converter.Setup(useOldXCResultExtractionMethod)
+	if !converter.Detect([]string{xcResultPath}) {
+		return nil, nil
+	}
+
+	results, err := xcresult3.ParseTestResults(xcResultPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse xcresult: %w", err)
+	}
+
+	testSummary, warnings, err := model3.Convert(results)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert xcresult data: %w", err)
+	}
+
+	if len(warnings) > 0 {
+		e.logger.Warnf("xcresult converter warnings:")
+		for _, warning := range warnings {
+			e.logger.Warnf("- %s", warning)
+		}
+	}
+
+	return testSummary, nil
+}
+
+func (e exporter) collectFlakyTestPlans(testSummary model3.TestSummary) []model3.TestPlan {
```

Comment:
If we ever move the xcresult parsing in go-xcode I can imagine having this and the `exportFlakyTestCases` function be implemented on the `model3.TestPlan` or other entities. That way the steps do not need to necessarily know how to calculate flakiness.